federation/pkg/federation-controller/namespace/namespace_controller.go

```diff
@@ -371,25 +413,21 @@ func (nc *NamespaceController) delete(namespace *api_v1.Namespace) error {
 		return fmt.Errorf("failed to delete events list from namespace: %v", err)
 	}
 
-	// Remove kube_api.FinalzerKubernetes
+	// Delete the namespace from all underlying clusters.
+	updatedNamespaceObj, err := nc.deletionHelper.ProcessDeletion(updatedNamespace)
+	if err != nil {
+		return err
+	}
+
+	// Remove kube_api.FinalizerKubernetes
```

Comment:
Yes it was blocked until the Delete request succeeded for all clusters but it wasnt waiting for all resources to go away. Updated the code to wait.